File: 1/1/4/CompoundOperators.java
/*
 * Activity 1.1.4
 */
// CODE TO ADD
/*
* TODO 1) Add at least three other people to your program,
* such as siblings, friends, or neighbors.
*
* TODO 2) Use people that are in different grades to validate
* your program can work for different school grades.
*
* TODO 3) For their years in school, use values that indicate
* they will soon be halfway through their current
* school year, as shown in the provided code.
*/
/*
* TODO 4)The average years you and your friends, siblings,
* and neighbors have been in school.
*
* TODO 5) The total days you have all been in school,
* assuming you spend 180 days per year in school.
*
* TODO 6) The average days you have all spent in school.
 */
public class CompoundOperators
{
  
  static class Person{
    String name;
    double years;

    Person(String name,double years){
        this.name = name;
        this.years = years;
    }
  }
  
  public static void main(String[] args)
  {
    int numPeople = 0;  
    double totalYears = 0;
    double avgYears = 0;
    int gradYear = 13;
    for (Person p: new Person[]{
        new Person("Kyle",12.5),
        new Person("Eric",11.5),
        new Person("Clara",8.5),
    }){

        numPeople++;
        totalYears +=p.years;
        avgYears = totalYears/numPeople;
        System.out.println(p.name + " has " + p.years + " years of school.\nthey have " +(gradYear-p.years)+ " years left of school");
    }
    System.out.println("number of people: " + numPeople);
    int totalDays = (int)(totalYears * 180);
    System.out.println("total days in school: " + totalDays);
    System.out.println("The average years of school is " + avgYears);
  }
}

File: 1/1/4/CompoundOperators.scala
case class Person(name: String,years: Double) // likely better more idomatic way to do this, tuple? but this is a java translation so idrc.
@main def main = 
    val gradYear = 13;
    val people = List(Person("kyle",12.5),Person("Eric",11.5),Person("Clara",8.5))
    val numPeople = people.length;
    val totalYears = people.map(_.years).sum;
    val avgYears = totalYears / numPeople;
    val totalDays: Int = (totalYears*180).toInt
    people.foreach(x => 
        println(s"${x.name} has ${x.years} of school.\nthey have ${gradYear-x.years} years left of school")
        )
    println(s"number of people: $numPeople")
    println(s"total days in school: $totalDays")
    println(s"The average years of school is $avgYears")